src/jsonapi.js

```diff
@@ -149,7 +150,7 @@ export const readEndpoint = (endpoint, {
         credentials: 'include'
       })
         .then(json => {
-          dispatch(apiRead({ endpoint, ...json }));
+          dispatch(apiRead({ endpoint, clearTypes, ...json }));
```

Comment:
I think we should add another argument, instead of "pollution" the payload w/"magic" properties. That should also be fairly straight forward? :)